b64_ntop and b64_pton

On Chimera Linux, the build fails with

```
signify.c:144:6: error: call to undeclared function 'b64_pton'; ISO C99 and
      later do not support implicit function declarations [-Wimplicit-function-declaration]
  144 |         if (b64_pton(commentend + 1, buf, buflen) != buflen)
      |             ^
signify.c:231:6: error: call to undeclared function 'b64_ntop'; ISO C99 and
      later do not support implicit function declarations [-Wimplicit-function-declaration]
  231 |         if (b64_ntop(buf, buflen, b64, sizeof(b64)) == -1)
      |             ^
signify.c:624:12: error: call to undeclared function 'b64_pton'; ISO C99 and
      later do not support implicit function declarations [-Wimplicit-function-declaration]
  624 |         if ((rv = b64_pton(hash, data, sizeof(data))) == -1)
      |                   ^
```
I see that somebody else had the same issue in #43, which seems stale. The proposed solution of adding the headers to `compat.h` works, but I got curious as to why it was working on a glibc system with the same compiler. It turns out that glibc's `resolv.h` has the same lineage as OpenBSD's and exports the relevant headers by chance, while musl's `resolv.h` does not. (Why it is that the base64 function headers are in `resolv.h` in the first place is a mystery to me.)

So that's why the build works on glibc systems even with `-Wimplicit-function-declaration`. Adding the headers to `compat.h` fixes this. I'll write a PR with the relevant changes.

--- a/compat.h
+++ b/compat.h
@@ -56,4 +56,10 @@ extern int bcrypt_pbkdf(const char *pass, size_t passlen,
 extern void explicit_bzero(void *buf, size_t len);
 #endif /* BUNDLED_BZERO */
 
+
+int b64_ntop(unsigned char const *src, size_t srclength, char *target,
+    size_t targsize);
+
+int b64_pton(char const *src, unsigned char *target, size_t targsize);
+
 #endif /* !COMPAT_H */

